### PR TITLE
Change EditAssignmentCommand and its parser for new feature design

### LINKED ISSUE #202: Overlapping functionality between grade command and edit_assignment command
The purpose of edit_assignment is to change a student's assignment score. This can be done by a grade command. Thus, it is worth considering whether to keep the grade command solely and remove edit_assignment.

### LINKED ISSUE #203: Redundant function of edit_assignment command
The edit_assignment command can be used to change the score of a non-submitted assignment, which could be considered unnecessary or even a bug.

### PR DESCRIPTION
**Pull request by jian7490**
Fix #202
Fix #203

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/EditAssignmentCommandParser.java [hunk 1]
```diff
@@ -24,25 +25,43 @@ public EditAssignmentCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argumentMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_STUDENT_INDEX, PREFIX_ASSIGNMENT_INDEX,
-                        PREFIX_ASSIGNMENT_SCORE);
-        if (!arePrefixesPresent(argumentMultimap, PREFIX_STUDENT_INDEX, PREFIX_ASSIGNMENT_INDEX,
-                PREFIX_ASSIGNMENT_SCORE)) {
+                        PREFIX_ASSIGNMENT_NAME, PREFIX_ASSIGNMENT_MAX_SCORE);
+        if (!arePrefixesPresent(argumentMultimap, PREFIX_STUDENT_INDEX, PREFIX_ASSIGNMENT_INDEX)) {
```
Comment (lm-44) [inline src/main/java/seedu/address/logic/parser/EditAssignmentCommandParser.java:29]: I assume that they are able to either change the assignmentName or maxScore or both at the same time, which is why there are only checks for student index and assignment index?

File: src/main/java/seedu/address/model/student/Student.java [hunk 2]
```diff
@@ -185,14 +186,20 @@ public Student addAssignment(Assignment assignmentToAdd) {
...
+    public Student updateAssignment(Index index, Assignment assignmentToUpdate)
+            throws AssignmentIndexOutOfRangeException {
+        int indexZeroBased = index.getZeroBased();
+        if (indexZeroBased < 0 || indexZeroBased >= assignmentList.size()) {
             throw new AssignmentIndexOutOfRangeException();
         }
-        assignmentList.remove(index - 1);
+        List<Assignment> newAssignmentList = new ArrayList<Assignment>(assignmentList);
+        newAssignmentList.set(indexZeroBased, assignmentToUpdate);
+        assignmentList = newAssignmentList;
```
Comment (lm-44) [inline src/main/java/seedu/address/model/student/Student.java:201]: Why do we have to create a newAssignmentList to set it? Should we just use the old assignment list to set?

File: src/test/java/seedu/address/logic/commands/EditAssignmentCommandTest.java
Comment (lm-44) [inline src/test/java/seedu/address/logic/commands/EditAssignmentCommandTest.java:1]: Great job on the tests! Personally, I have a huge trouble in making them in this project.

**Review by lm-44 — COMMENTED** [review]
LGTM! Just seeking some clarifications on parts of the code.

File: src/main/java/seedu/address/logic/parser/EditAssignmentCommandParser.java (hunk 1 above)
Comment (jian7490) [reply src/main/java/seedu/address/logic/parser/EditAssignmentCommandParser.java:29]: Great observation! This if condition here is intended to check missing prefixes, and only the two indices are must-have prefixes. As you said, the assignmentName and maxScore are optional to be changed.

File: src/main/java/seedu/address/model/student/Student.java (hunk 2 above)
Comment (jian7490) [reply src/main/java/seedu/address/model/student/Student.java:201]: Good point. I will fix this.

**Pull request merged**